src/fileActions.jsx

```diff
@@ -603,3 +603,30 @@ export const updateFile = (file, currentPath) => {
                     return file;
             }, exc => console.error(`Adding file ${file} failed: ${exc.toString()}`));
 };
+
+export const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
+        external: {
+            "content-disposition": `attachment; filename="${selected.name}"`,
+            "content-type": "application/octet-stream",
+            "content-length": Number(selected.size),
```

Comment:
I thought setting `content-length` wouldn't work, and instead this is blocked on the cockpit PR?

Reply:
Good point, we should drop this. Cockpit-ws should handle this transparently now.